url7a45f192ca_mfullen_ping_pong_net_tgz-pJ8-ping_pong_net_client_io_TestClientNonPPNServerJ8/src/ping/pong/net/server/ClientExample.java

```diff
@@ -0,0 +1,302 @@
+            System.setProperty("javax.net.ssl.keyStorePassword", ConnectionConfiguration.DEFAULT_KEY_STORE_PASSWORD);
+            System.setProperty("java.protocol.handler.pkgs", "com.sun.net.ssl.internal.www.protocol");
+
+            String absolutePath = new File(Thread.currentThread().getContextClassLoader().getResource("pingpong.crt").toURI()).getAbsolutePath();
+            System.out.println(absolutePath);
+            System.setProperty("javax.net.ssl.trustStore", absoluteKeyPath);
+            System.setProperty("javax.net.ssl.trustStorePassword", "pingpong123");
+
+            // SSLSocketFactory factory = (SSLSocketFactory) ctx.getSocketFactory();
+            SSLSocketFactory factory = (SSLSocketFactory) SSLSocketFactory.getDefault();
+            // SSLSocket client = (SSLSocket) factory.createSocket("localhost", 5011);
+            Socket client = (SSLSocket) factory.createSocket("localhost", 5011);
+            //  client.startHandshake();
+            // java.security.cert.Certificate[] serverCerts = client.getSession().getPeerCertificates();
+            ObjectInputStream inputStream = new ObjectInputStream(client.getInputStream());
+            int readInt = inputStream.readInt();
+            System.out.println("SSL- I was assigned Client id: " + readInt);
+            ObjectOutputStream outputStream = new ObjectOutputStream(client.getOutputStream());
+            outputStream.writeObject(new byte[]
+                    {
+                        1, 3, 2, 3
+                    });
+            outputStream.flush();
+            outputStream.close();
+        }
+    }
+
+    public static void ssl2() throws IOException,
+                                     InterruptedException,
+                                     NoSuchAlgorithmException,
+                                     KeyManagementException,
+                                     KeyStoreException,
+                                     UnrecoverableKeyException,
+                                     CertificateException,
+                                     URISyntaxException
+    {
+        {
+
+
+
+            // SSLSocketFactory factory = HttpsURLConnection.getDefaultSSLSocketFactory();
+
+
+            KeyStore ks = KeyStore.getInstance("JKS");
+            String absolutePath = new File(Thread.currentThread().getContextClassLoader().getResource(ConnectionConfiguration.DEFAULT_KEY_STORE).toURI()).getAbsolutePath();
+            ks.load(new FileInputStream(absolutePath), (ConnectionConfiguration.DEFAULT_KEY_STORE_PASSWORD).toCharArray());
+            TrustManagerFactory tmf = TrustManagerFactory.getInstance("SunX509");
+            tmf.init(ks);
+            SSLContext sslcontext = SSLContext.getInstance("SSLv3");
+            sslcontext.init(null, tmf.getTrustManagers(), null);
+
+            SSLSocketFactory factory = (SSLSocketFactory) sslcontext.getSocketFactory();
+            SSLSocket client = (SSLSocket) factory.createSocket("localhost", 5011);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *client*. The resource is referenced at lines 132, 133, 134, 136, 137, 139, 140, 144. The resource is closed at lines 145, 147. There are other execution paths that do not contain closure statements, for example, when ObjectOutputStream.writeObject throws an exception. Ensure *client* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).